subprojects/docs/src/snippets/userguide/dependencyManagement/customizingResolution/ivyMetadataRule/kotlin/build.gradle.kts

```diff
@@ -15,6 +15,11 @@ open class IvyVariantDerivationRule : ComponentMetadataRule {
     @Inject open fun getObjects(): ObjectFactory = throw UnsupportedOperationException()
 
     override fun execute(context: ComponentMetadataContext) {
+        // This filters out any non Ivy module
+        if(context.getDescriptor(IvyModuleDescriptor) == null) {
```

Comment:
This is not valid in Kotlin. You have to use `context.getDescriptor(IvyModuleDescriptor::class)`

Reply:
oops! nice catch. fixed in b1b94ef